### PR TITLE
BigQuery: sample for writing query results to a destination table

### PR DESCRIPTION
**Pull request by tswast**
Several people have asked for this sample, finally making one for the official docs.

### CODE REVIEW

File: docs/bigquery/snippets.py [hunk 1]
```diff
@@ -697,6 +697,54 @@ def test_client_query(client):
...
+def test_client_query_destination_table(client, to_delete):
+    """Run a query"""
+    dataset_id = 'query_destination_table_{}'.format(_millis())
+    dataset_ref = client.dataset(dataset_id)
+    to_delete.append(dataset_ref)
+    dataset = client.create_dataset(bigquery.Dataset(dataset_ref))
+    to_delete.insert(0, dataset_ref.table('your_table_id'))
+
+    # [START bigquery_query_destination_table]
+    from google.cloud.bigquery import QueryJobConfig
```
Comment (jonparrott) [inline docs/bigquery/snippets.py]: Import modules, not members?

File: docs/bigquery/snippets.py [hunk 2]
```diff
@@ -697,6 +697,54 @@ def test_client_query(client):
...
+    # The write_disposition specifies the behavior when writing query results
+    # to a table that already exists. With WRITE_TRUNCATE, any existing rows
+    # in the table are overwritten by the query results.
+    job_config.write_disposition = 'WRITE_TRUNCATE'
+
+    # Start the query, passing in the extra configuration.
+    query_job = client.query(
+        'SELECT 17 AS my_col;', job_config=job_config)
+
+    # Waits for the query to finish
```
Comment (jonparrott) [inline docs/bigquery/snippets.py]: You can just iterate through the future now.

Comment (tswast) [reply docs/bigquery/snippets.py]: Done.

File: docs/bigquery/snippets.py (hunk 1 above)
Comment (tswast) [reply docs/bigquery/snippets.py]: Done.

File: docs/bigquery/snippets.py [hunk 3]
```diff
@@ -697,6 +697,48 @@ def test_client_query(client):
     # [END client_query]
 
 
+def test_client_query_destination_table(client, to_delete):
+    """Run a query"""
+    dataset_id = 'query_destination_table_{}'.format(_millis())
+    dataset_ref = client.dataset(dataset_id)
+    to_delete.append(dataset_ref)
+    dataset = client.create_dataset(bigquery.Dataset(dataset_ref))
```
Comment (dhermes) [inline docs/bigquery/snippets.py]: @tswast `flake8` is sad about this unused variable (`dataset`)

Comment (tswast) [reply docs/bigquery/snippets.py]: Ah, oops. I bumped the to_delete up and forgot to remove that. It can go.

**Pull request merged**